File: backend/src/main/java/com/sarahjewellers/model/EmailTemplate.java
package com.sarahjewellers.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "email_templates")
@Data
public class EmailTemplate {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String occasion; // BIRTHDAY, ANNIVERSARY, FESTIVAL

    private String subject;

    @Column(columnDefinition = "TEXT")
    private String body;

    private String festivalName; // for FESTIVAL type (Diwali, Ramadan, etc.)
}

File: backend-node/models/EmailTemplate.js
const mongoose = require('mongoose');

const emailTemplateSchema = new mongoose.Schema({
  occasion:     { type: String, required: true, enum: ['BIRTHDAY', 'ANNIVERSARY', 'FESTIVAL'] },
  festivalName: { type: String },
  subject:      { type: String, required: true },
  body:         { type: String, required: true }
}, { timestamps: true });

module.exports = mongoose.model('EmailTemplate', emailTemplateSchema);
